lib/src/ui/movie_list.dart

```diff
@@ -63,54 +64,51 @@ class _MovieListState extends State<MovieList> {
           ),
         ),
       ),
-      body: Container(
-        decoration: BoxDecoration(
-          gradient: backgroundColor,
-        ),
-        child: StreamBuilder(
-          stream: widget.movieBloc.allMovies,
-          builder: (context, AsyncSnapshot<Movie> snapshot) {
-            if (snapshot.hasData) {
-              return buildList(snapshot.data!);
-            } else if (snapshot.hasError) {
-              return Text(
-                snapshot.error.toString(),
-              );
-            }
-            return Center(
-              child: CircularProgressIndicator(),
+      body: StreamBuilder(
+        stream: widget.movieBloc.allMovies,
+        builder: (context, AsyncSnapshot<Movie> snapshot) {
+          if (snapshot.hasData) {
+            return buildList(snapshot.data!);
```

Comment:
You have to type the StreamBuilder like `StreamBuilder<List<Movie>>` in order to validate the type of the stream as well and the data object as well, if not types are assumed and thats not good.

Reply:
Fixed